src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
but you are correct about subject buffer the workspace belong to can be changed. so, the code still has a bug (I will open a bug). the tagger should subscribe to WorkspaceRegistration.WorkspaceChanged event and handle those case. (this might be common issue in tagger. not sure)